thunx-model/src/main/java/com/contentgrid/thunx/predicates/model/CollectionValue.java

```diff
@@ -0,0 +1,28 @@
+package com.contentgrid.thunx.predicates.model;
+
+import com.contentgrid.opa.rego.ast.Term;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+import java.util.Collection;
+
+@EqualsAndHashCode
+public class CollectionValue implements Scalar<Collection<Term.ScalarTerm<?>>> {
```

Comment:
`thunx-model` should not have a dependency on opa(-java-client). The intention of this thunx-model is to make an abstraction of any policy engine syntax. Should this implement `Scalar<Collection<Scalar<?>>>` instead ?

Additionally, I think it should probably be possible to have a non-scalar in a collection (for example a variable) ? I'm not sure a _collection_ should only contain scalars, so maybe this should implement something like `FunctionExpression<Collection<?>>` instead ?